opentelemetry-sdk/src/opentelemetry/sdk/metrics/export/batcher.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+from typing import Sequence, Type
+
+from opentelemetry.metrics import Counter, MetricT
+from opentelemetry.sdk.metrics.export import MetricRecord
+from opentelemetry.sdk.metrics.export.aggregate import (
+    Aggregator,
+    CounterAggregator,
+)
+
+
+class Batcher(abc.ABC):
+    """Base class for all batcher types.
+
+    The batcher is responsible for storing the aggregators and aggregated
+    values received from updates from metrics in the meter. The stored values will be
+    sent to an exporter for exporting.
+    """
+
+    def __init__(self, keep_state: bool):
+        self._batch_map = {}
+        # keep_state=True indicates the batcher computes checkpoints from over
+        # the process lifetime. False indicates the batcher computes
+        # checkpoints which describe the updates of a single collection period
+        # (deltas)
+        self.keep_state = keep_state
+
+    @property
+    def batch_map(self):
+        return self._batch_map
+
+    @batch_map.setter
+    def batch_map(self, value):
```

Comment:
type of the value would be useful here, so people understand this a dictionary of tuples (aggregator, label_set)